src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -684,46 +715,24 @@ public bool TryAppendEntireSegmentFromSmuggler(DocumentsOperationContext context
             TimeSeriesValuesSegment segment,
             DateTime baseline)
         {
-
             if (IsOverlapping(context, key, collectionName, segment, baseline))
                 return false;
 
             // if this segment isn't overlap with any other we can put it directly
-            ValidateSegment(segment);
-            var table = GetOrCreateTimeSeriesTable(context.Transaction.InnerTransaction, collectionName);
             using (var holder = new TimeSeriesSegmentHolder(this, context, documentId, name, collectionName, fromReplicationChangeVector: changeVector, timeStamp: baseline))
             {
-                var slicer = holder.SliceHolder;
-
-                Stats.UpdateStats(context, slicer, collectionName, segment, baseline, segment.NumberOfLiveEntries);
-
-                var newEtag = _documentsStorage.GenerateNextEtag();
-                changeVector ??= _documentsStorage.GetNewChangeVector(context, newEtag);
-
-                _documentDatabase.TimeSeriesPolicyRunner?.MarkSegmentForPolicy(context, slicer, baseline, changeVector, segment.NumberOfLiveEntries);
-
-                if (segment.NumberOfLiveEntries == 0)
+                if (holder.LoadCurrentSegment())
                 {
-                    MarkSegmentAsPendingDeletion(context, collectionName.Name, newEtag);
+                    // should never happen 
+                    return false;
```

Comment:
if this should never happened let's also add a `Debug.Assert` here.

Reply:
Addressed in https://github.com/ravendb/ravendb/pull/12982/commits/a09d2ba0f9f5aba8840884deea9329d98f0f0562